src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -266,5 +273,33 @@ protected virtual string GenerateUrl(string protocol, string host, VirtualPathDa
                 builder.Clear();
             }
         }
+
+        private string FastGetUrl(
```

Comment:
Do we have adequate tests for this new code path? I suspect we could stand to add a few